cli/npm/registry.rs

```diff
@@ -397,4 +409,52 @@ impl CliNpmRegistryApiInner {
       None
     }
   }
+
+  // Search is only used by lsp import completion, so caching etc. is simpler.
+  pub async fn search(
+    &self,
+    query: &str,
+  ) -> Result<Arc<Vec<String>>, AnyError> {
+    if let Some(names) = self.search_cache.lock().get(query) {
+      return Ok(names.clone());
+    }
+
+    #[derive(Debug, Deserialize)]
+    struct Package {
+      name: String,
+    }
+    #[derive(Debug, Deserialize)]
+    struct Object {
+      package: Package,
+    }
+    #[derive(Debug, Deserialize)]
+    struct Response {
+      objects: Vec<Object>,
+    }
+
+    let mut search_url = self.base_url.clone();
+    search_url
+      .path_segments_mut()
+      .map_err(|_| anyhow!("Custom npm registry URL cannot be a base."))?
+      .pop_if_empty()
+      .extend("-/v1/search".split('/'));
+    search_url
+      .query_pairs_mut()
+      .append_pair("text", &format!("{} boost-exact:false", query));
+
+    let bytes = self.http_client.download(search_url).await?;
```

Comment:
In the previous commit this was using `ModuleRegistry`'s file fetcher. Is it ok for this to be using the http client directly here (I think it's probably ok?)? If not, maybe we should just move this functionality out of `CliNpmRegistryApi` and make a more targeted `NpmSearchApi` for the lsp in order to not accidentally mix the two together and reduce the dependencies necessary to create a `CliNpmRegistryApi`.